### PR TITLE
Add code coverage

### PR DESCRIPTION
**Pull request by MPParsley**
- Re-enabled xdebug by installing Drupal one directory up. (due to a bug during module lookup that causes the stack to grow too much)
- Added phpunit test reporter

Code coverage allows us to visualize progress and to enforce unit tests in new features.
See demo on https://youtu.be/r1joerjQL_c?t=33

This PR only reports code coverage to code climate. Code coverage can be enforced with GitHub's branch protection and configured in:
Schermafbeelding 2020-02-08 om 19 06 34

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -81,4 +76,12 @@ before_script:
...
 
+  # Get the CodeClimate test reporter.
+  - curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
+  - chmod +x ./cc-test-reporter
+  - ./cc-test-reporter before-build
+
 script: DRUPAL_DIR=$DRUPAL_DIR MODULE_DIR=$MODULE_DIR $MODULE_DIR/scripts/travis-ci/run-test.sh $TEST_SUITE
+
+after_script:
+  - if [ "$TRAVIS_PULL_REQUEST" == "false" ] && [ "${TEST_SUITE}" != "PHP_CodeSniffer" ]; then ./cc-test-reporter after-build --coverage-input-type clover --exit-code $TRAVIS_TEST_RESULT; fi
```
Comment (MPParsley) [inline .travis.yml]:
```suggestion
  - if [ "$TRAVIS_PULL_REQUEST" == "false" ] && [ "${TEST_SUITE}" != "PHP_CodeSniffer" ]; then ./cc-test-reporter after-build --exit-code $TRAVIS_TEST_RESULT; fi
```

**Comment by pfrenssen** [comment]
I'm not sure that this is possible with Drupal, the code base is too large to be processed in a timely fashion. I am also seeing this when trying to run code coverage analysis on my local machine.

Do you know of any examples of Drupal modules that have managed to get this working?

**Comment by MPParsley** [comment]
We have some projects where we test code coverage none of them has as many Kernel tests as og.

Og has a lot of API and it will probably take too long to run, so I'm fine with closing this issue.

**Comment by MPParsley** [comment]
@pfrenssen , fyi : https://github.com/mkalkbrenner/search_api_solr/pull/22#issuecomment-625397661

**Comment by pfrenssen** [comment]
Github Actions are indeed a possibility to look into.

**Pull request closed without merging**